stetho/src/main/java/com/facebook/stetho/inspector/protocol/module/DOM.java

```diff
@@ -208,6 +216,103 @@ public void run() {
     });
   }
 
+  @ChromeDevtoolsMethod
+  public PerformSearchResponse performSearch(JsonRpcPeer peer, final JSONObject params) {
+    return mDOMProvider.postAndWait(new UncheckedCallable<PerformSearchResponse>() {
+      @Override
+      public PerformSearchResponse call() {
+        final PerformSearchRequest request = mObjectMapper.convertValue(
+            params,
+            PerformSearchRequest.class);
+
+        // Searches are case insensitive so convert
+        // the query to lower case for simpler comparisons.
+        final String lowerCaseQuery = request.query.toLowerCase();
+
+        final Object root = mShadowDOM.getRootElement();
+
+        // Each search action has a unique ID so that
+        // it can be queried later.
+        final String id = UUID.randomUUID().toString();
```

Comment:
As an aside, you can pick up an interesting perf win by simply addressing an array instead of using a map here:

  synchronized (...) {
    String id = String.valueOf(mSearchResults.size());
    mSearchResults.add(results);
  }

Then when the request id comes back in you'd simply do:

  int index = Integer.parseInt(requestId);
  synchronized (...) {
    ManagedIntArray results = mSearchResults.get(index);
  }